generator/src/lib.rs

```diff
@@ -1989,83 +1980,103 @@ fn derive_setters(
             });
         }
 
-        // TODO: Improve in future when https://github.com/rust-lang/rust/issues/53667 is merged id:6
-        if field.reference.is_some() {
-            if field.basetype == "char" && matches!(field.reference, Some(vkxml::ReferenceType::Pointer)) {
+        if field.basetype == "char" {
+            let param_ident_as_c_str = format_ident!("{}_as_c_str", param_ident_short);
+            if matches!(field.reference, Some(vkxml::ReferenceType::Pointer)) {
                 assert!(field.null_terminate);
                 assert_eq!(field.size, None);
-                return Some(quote!{
+                return Some(quote! {
                     #[inline]
                     #deprecated
-                    pub fn #param_ident_short(mut self, #param_ident_short: &'a ::std::ffi::CStr) -> Self {
+                    pub fn #param_ident_short(mut self, #param_ident_short: &'a std::ffi::CStr) -> Self {
                         self.#param_ident = #param_ident_short.as_ptr();
                         self
                     }
+                    #[inline]
+                    #deprecated
+                    pub unsafe fn #param_ident_as_c_str(&self) -> &std::ffi::CStr {
+                        std::ffi::CStr::from_ptr(self.#param_ident)
```

Comment:
@Ralith I don't remember if we discussed this but these pointers can trivially be `NULL`. Fortunately the function is `unsafe` and this call is _not_ used in `Debug`, so I'm curious if it was deliberate.

However, we might want to switch this to return `Option` to represent that (and to safely use this API in `Debug`.

Reply:
Oops. I agree we should return `Option` here, and further, we should probably accept `Option` in the setters.

Reply:
> we should probably accept `Option` in the setters

Same for the pointer slice setters.

> I don't think we can safely use this API in `Debug` regardless because the pointer might be garbage.

Exactly, that's also what I wrote in https://github.com/ash-rs/ash/pull/858#issue-2075037358: we can only print (subslices of) static arrays that are embedded in known-valid structs.

Reply:
> Same for the pointer slice setters.

And arbitrary struct pointers (if Vulkan denotes them optional). Perhaps we should ask ourselves if the builder pattern isn't intended to (re)set fields to `NULL`?

Reply:
Good point. Let's not. The asymmetry is weird, but treating slices specially is weirder, and putting `Option` in every setter would be a significant downgrade in ergonomics.

Reply:
> The asymmetry is weird

On that note, how about providing similar `as_slice()` helpers for the pointer-slice builders?

Reply:
Seems reasonable, though of limited use outside implementing layers or ICDs.

Reply:
In hindsight it makes total sense, as the caller already allocates and passes those mutable arrays/sizes. That's not the case for the static-arrays in #858, so they're the only ones with a special case.

Reply:
I'm confused -- are you saying it makes sense to not have getters for pointer-slice fields? Why would the storage not being inline make that any less useful?

Reply:
As you said, it's of limited use outside of layers and ICDs. Might be a useful addition if we wish to target those.

Just thinking out loud why I've never needed these slices helpers in client code :wink:

Reply:
Ah, yeah. I know there's been occasional interest. We could have a feature gate for implementer-side APIs, even. Still, far from urgent.